src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -411,58 +410,61 @@ private[collection] object HashTable {
-      // k *= 0x5bd1e995
-      // k
-
-      /* Another fast multiplicative hash
-       * by Phil Bagwell
-       *
-       * Comment:
-       * Multiplication doesn't affect all the bits in the same way, so we want to
-       * multiply twice, "once from each side".
-       * It would be ideal to reverse all the bits after the first multiplication,
-       * however, this is more costly. We therefore restrict ourselves only to
-       * reversing the bytes before final multiplication. This yields a slightly
-       * worse entropy in the lower 8 bits, but that can be improved by adding:
-       *
-       * `i ^= i >> 6`
-       *
-       * For performance reasons, we avoid this improvement.
-       * */
-      val i= scala.util.hashing.byteswap32(hcode)
-
-      /* Jenkins hash
-       * for range 0-10000, output has the msb set to zero */
-      // var h = hcode + (hcode << 12)
-      // h ^= (h >> 22)
-      // h += (h << 4)
-      // h ^= (h >> 9)
-      // h += (h << 10)
-      // h ^= (h >> 2)
-      // h += (h << 7)
-      // h ^= (h >> 12)
-      // h
-
-      /* OLD VERSION
-       * quick, but bad for sequence 0-10000 - little entropy in higher bits
-       * since 2003 */
-      // var h: Int = hcode + ~(hcode << 9)
-      // h = h ^ (h >>> 14)
-      // h = h + (h << 4)
-      // h ^ (h >>> 10)
-
-      // the rest of the computation is due to SI-5293
-      val rotation = seed % 32
-      val rotated = (i >>> rotation) | (i << (32 - rotation))
-      rotated
+    /** Murmur hash
+     * {{{
+     *  m = 0x5bd1e995
+     *  r = 24
+     *  note: h = seed = 0 in mmix
+     *  mmix(h,k) = k *= m; k ^= k >> r; k *= m; h *= m; h ^= k;
+     *
+     *  var k = hcode * 0x5bd1e995
+     *  k ^= k >> 24
+     *  k * 0x5bd1e995
+     * }}}
+     *
+     * Another fast multiplicative hash
+     * by Phil Bagwell
+     *
+     * Comment:
+     * Multiplication doesn't affect all the bits in the same way, so we want to
+     * multiply twice, "once from each side".
+     * It would be ideal to reverse all the bits after the first multiplication,
+     * however, this is more costly. We therefore restrict ourselves only to
+     * reversing the bytes before final multiplication. This yields a slightly
+     * worse entropy in the lower 8 bits, but that can be improved by adding:
+     * {{{
+     *  i ^= i >> 6
+     * }}}
+     * For performance reasons, we avoid this improvement.
+     *
+     * Jenkins hash
+     * for range 0-10000, output has the msb set to zero
+     * {{{
+     *  var h = hcode + (hcode << 12)
+     *  h ^= (h >> 22)
+     *  h += (h << 4)
+     *  h ^= (h >> 9)
+     *  h += (h << 10)
+     *  h ^= (h >> 2)
+     *  h += (h << 7)
+     *  h ^  (h >> 12)
+     * }}}
+     * OLD VERSION
+     * quick, but bad for sequence 0-10000 - little entropy in higher bits
+     * since 2003
+     * {{{
+     * var h: Int = hcode + ~(hcode << 9)
+     * h = h ^ (h >>> 14)
+     * h = h + (h << 4)
+     * h ^ (h >>> 10)
+     * }}}
```

Comment:
This part is important: a fast and small algorithm that we actually used but in practice works poorly.  "Don't do this!"  Leave this in.